File: records/choices.py
"""
A central repo for choices for various fields of forms
"""


PROGRAM_LEVEL_CHOICES = (
    ("be", "Bachelors"),
    ("msc", "Masters"),
    ("phd", "PhD")
)


BE_PROGRAM_CHOICES = (
    ("BCT", "BE Computer (BCT)"),
    ("BEX", "BE Electronics (BEX)"),
    ("BEI", "BEI"),
)

MSC_PROGRAM_CHOICES = (
    ("MSICE", "MSc in Information & Communications Engineering (MScICE)"),
    ("MSCSK", "MSc in Computer Systems and Knowledge engineering (MScCKSE)"),
)


GROUPED_PROGRAM_CHOICES = (
    ('Bachelors', BE_PROGRAM_CHOICES),
    ('Masters', MSC_PROGRAM_CHOICES),
    ('PhD', (
        ('PhD', 'PhD'),
    )),
)

BE_STUDENT_GROUP_CHOICES = (
    ('A', 'A'),
    ('B', 'B'),
    ('C', 'C'),
    ('D', 'D'),
)

TITLE_CHOICES = (
    ("Dr.", "Dr."),
    ("Er.", "Er."),
    ("Mr.", "Mr."),
    ("Ms.", "Ms."),
    ("Mrs.", "Mrs."),
)

BE_PROGRAM_TYPE_CHOICES = (
    ('Regular', 'Regular'),
    ('Full Fee', 'Full Fee'),
)

MSC_PROGRAM_TYPE_CHOICES = (
    ('Regular', 'Regular'),
    ('Full Fee', 'Full Fee'),
    ('Sponsorship', 'Sponsorship'),
)

GENDER_CHOICES = (
    ('Male', 'Male'),
    ('Female', 'Female'),
    ('Other', 'Other'),
)

ADDRESS_TYPE_CHOICES = (
    ("Current", "Current"),
    ("Permanent", "Permanent"),
)

ACADEMIC_LEVEL_CHOICES = (
    ("Masters", "Masters"),
    ("PhD", "PhD"),
    ("Post Doc.", "Post Doc."),
)

ACADEMIC_STATUS_CHOICES = (
    ("Ongoing", "Ongoing"),
    ("Completed", "Completed"),
)

EMPLOYMENT_STATUS_CHOICES = (
    ("Employed", "Employed"),
    ("Unemployed", "Unemployed"),
)

File: institutuff/static/institutuff/choices.js
hierarchy = ['years','levels','programs','groups']
/////////////////////////////////////////programs
bct_program = {
    'into' : false,
    'tag' : 'groups',
    'value' : 'bct',
    'name' : 'BE Computer(BCT)',
    'members' :[
        {'value':'A'},{'value':'B'},{'value':'C'},{'value':'D'}
    ]
} 
bex_program = {
    'into' : false,
    'tag' : 'groups',
    'value' : 'bex',
    'name' : 'BE Electronics(BEX)',
    'members' :[
        {'value':'A'},{'value':'B'},{'value':'C'},{'value':'D'}
    ]
} 
bei_program = {
    'into' : false,
    'tag' : 'groups',
    'value' : 'bei',
    'name' : 'BEI',
    'members' :[
        {'value':'A'},{'value':'B'},{'value':'C'},{'value':'D'}
    ]
}
msice_program = {
    'into': false,
    'tag': 'groups',
    'value': 'MSICE',
    'name': 'MSc in Information & Communications Engineering (MScICE)',
    'members': [
        { 'value': 'A' }, { 'value': 'B' }, { 'value': 'C' }, { 'value': 'D' }
    ]
    }
mscsk_program = {
    'into': false,
    'tag': 'groups',
    'value': 'MSCSK',
    'name': 'MSc in Computer Systems and Knowledge engineering (MScCKSE)',
    'members': [
        { 'value': 'A' }, { 'value': 'B' }, { 'value': 'C' }, { 'value': 'D' }
    ]
}
//////////////////////////////////////////////////////////levels
bachelors_level = {
    'into' : true,
    'tag' : 'programs',
    'value':'bachelors',
    'members':[
        bct_program,bex_program,bei_program                                       
    ]
}
masters_level = {
    'into': true,
    'tag': 'programs',
    'value': 'masters',
    'members': [
        msice_program, mscsk_program
    ]
}
//////////////////////////////////////////////////////////years
year_2075 = {
    'into' : true,
    'tag' : 'levels',
    'value': '2075',
    'members': [
        bachelors_level, masters_level
            ]
}
year_2074 = {
    'into': true,
    'tag': 'levels',
    'value': '2074',
    'members': [
        bachelors_level, masters_level
    ]
}
////////////////////////////////////////////////////////
json_obj={
    'into' : true,
    'tag' : 'years',
    'members' : [   
        year_2075, year_2075
    ]
}